qiita_ware/ebi.py

```diff
@@ -77,56 +80,74 @@ def iter_file_via_list_of_dicts(input_file):
     preprocessed_data_id : str
-    study_title : str
-    study_abstract : str
-    investigation_type : str
-        Any of the options provided by ebi, see:
-        https://www.ebi.ac.uk/ega/sites/ebi.ac.uk.ega/files/documents/Study.xml
-        If `'Other'` is passed, then you must provide a value in the
-        new_investigation_type argument.
-    empty_value : str, optional
-        Defaults to "no_data". This is the value that will be used when data
-        for a particular metadata field is missing
-    new_investigation_type: str, optional
-        'metagenome', and 'mimarks-survey' are specially recognized and used to
-        set other attributes in the submission, but any string is valid. This
-        value is required only if `'Other'` is passed in `investigation_type`.
     """
-    def __init__(self, preprocessed_data_id, study_title, study_abstract,
-                 investigation_type, empty_value='no_data',
-                 new_investigation_type=None, pmids=None, **kwargs):
+    def __init__(self, preprocessed_data_id, action, **kwargs):
+        self.valid_ebi_actions = ('ADD', 'VALIDATE', 'MODIFY')
+        self.valid_ebi_submission_states = ('submitting', 'success')
+
+        # Step 1: variable setup and validations
+        if action not in self.valid_ebi_actions:
+            raise ValueError("Not a valid action (%s): %s" % (
+                ', '.join(self.valid_ebi_actions), action))
+
+        ena_ontology = Ontology(convert_to_id('ENA', 'ontology'))
+        ppd = PreprocessedData(preprocessed_data_id)
+        s = Study(ppd.study)
+        pt = PrepTemplate(ppd.prep_template)
+
+        status = ppd.submitted_to_insdc_status()
+        if status in self.valid_ebi_submission_states:
+            raise ValueError("Cannot resubmit! Current status is: %s" % status)
+
         self.preprocessed_data_id = preprocessed_data_id
-        self.study_title = study_title
-        self.study_abstract = study_abstract
-        self.investigation_type = investigation_type
-        self.empty_value = empty_value
-        self.new_investigation_type = new_investigation_type
+        self.study_title = s.title
+        self.study_abstract = s.info['study_abstract']
+
+        it = pt.investigation_type
+        key = it
+        if it in ena_ontology.terms:
+            self.investigation_type = it
+            self.new_investigation_type = None
+        elif it in ena_ontology.user_defined_terms:
+            self.investigation_type = 'Other'
+            self.new_investigation_type = it
+        else:
+            # This should never happen
+            raise ValueError("Unrecognized investigation type: '%s'. This "
+                             "term is neither one of the official terms nor "
+                             "one of the user-defined terms in the ENA "
+                             "ontology")
+        ts = datetime.now().strftime('%Y_%m_%d_%H:%M:%S')
+        self.ebi_dir = '%s_%s' % (preprocessed_data_id, ts)
```

Comment:
probably better to also include a string that denotes that this is a EBI submission:
```python
self.ebi_dir = "ebi_submission_%s_%s" % (preprocessed_data_id, ts)
```